### PR TITLE
Add rendering performance benchmark page

### PR DESCRIPTION
**Pull request by anandthakker**

### CODE REVIEW

File: benchmark/render.html [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+      const results = [];
+
+      async function run(latex) {
+        const div = document.createElement('div');
+        div.classList.add('test-case');
+        div.innerHTML = `
+            <div class="test-case-render">
+              <div class="mathquill-math-field"></div>
+              <div class="html-only mq-math-mode"></div>
```
Comment (jwmerrill) [inline benchmark/render.html:86]: This is a tiny thing, but I think MQ marks its root block as `display: inline` or maybe `inline-block`, so these end up on the same line. An easy fix would be making these spans, and then wrapping each of them in a parent div.

**Comment by jwmerrill** [comment]
I tried adding the following to force the browser to do synchronous layout to try to make the comparison to html a little more fair:
```
> git diff
diff --git a/benchmark/render.html b/benchmark/render.html
index 7d861cd8..9e98b20b 100644
--- a/benchmark/render.html
+++ b/benchmark/render.html
@@ -97,11 +97,15 @@

         let start = performance.now();
         MQ.MathField(mqElement);
+        // Force syncronous layout
+        document.body.getBoundingClientRect();
         const mqDuration = performance.now() - start;

         const html = mqElement.innerHTML;
         start = performance.now();
         htmlElement.innerHTML = html;
+        // Force syncronous layout
+        document.body.getBoundingClientRect();
         const htmlDuration = performance.now() - start;
         results.push({latex, mqDuration, htmlDuration});
```
Found that it took typical html times from 0-1ms to more like 2-5ms.

File: benchmark/render.html [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          const arg = randomLatex(depth - 1);
+          length += arg.length;
+          if (length + arg.length > MAX_LENGTH) break;
+          args.push(arg);
+        }
+        return fn(...args);
+      }
+
+      function randomInteger(maxDigits) {
+        return Math.floor((random() * 10) ^ maxDigits);
```
[the remark this answers is not in the document]
Comment (anandthakker) [reply benchmark/render.html:150]: LOL yes... yes it should

**Comment by anandthakker** [comment]
@jwmerrill updated to address your notes

**Comment by anandthakker** [comment]
(Rebased on to `main`, but only the last two commits are new)

**Review by jwmerrill — APPROVED** [review]

**Pull request merged**